cpp/src/io/parquet/experimental/hybrid_scan_preprocess.cu

```diff
@@ -312,23 +311,26 @@ hybrid_scan_reader_impl::prepare_dictionaries(
 
 void hybrid_scan_reader_impl::update_row_mask(cudf::column_view in_row_mask,
                                               cudf::mutable_column_view out_row_mask,
+                                              cudf::size_type out_row_mask_offset,
                                               rmm::cuda_stream_view stream)
 {
   CUDF_FUNC_RANGE();
 
   auto const total_rows = static_cast<cudf::size_type>(in_row_mask.size());
 
-  CUDF_EXPECTS(total_rows == out_row_mask.size(),
+  CUDF_EXPECTS(out_row_mask_offset + total_rows <= out_row_mask.size(),
```

Comment:
We are updating only a portion (spanned by current output table chunk size) of the output row mask from the input row mask